backend/src/v5/processors/teamspaces/projects/models/federations.js

```diff
@@ -62,7 +64,17 @@ Federations.deleteFavourites = async (username, teamspace, project, favouritesTo
 	await deleteFavourites(username, teamspace, accessibleFederations, favouritesToRemove);
 };
 
-Federations.newRevision = queueFederationUpdate;
+Federations.newRevision = async (teamspace, project, federation, info) => {
+	const revisionId = generateUUID();
+	const revision = {
+		_id: revisionId,
+		author: info.owner,
+		timestamp: new Date(),
+		containers: info.containers,
+	};
+	await createRevision(teamspace, project, federation, revision); // Should await on this because updateModelSubModels is going to emit an event which will expect a revision node
```

Comment:
this function already exists in v5 as we had to create a revision entry for drawings (the function itself probably needs to be altered to cater for federations!) https://github.com/3drepo/3drepo.io/blob/master/backend/src/v5/models/revisions.js#L117